### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Shevtsooov**
[DEMO LINK](https://Shevtsooov.github.io/react_autocomplete/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,57 +1,63 @@
...
-      </div>
+      <Dropdown
+        query={query}
+        setQuery={setQuery}
+        applyQuery={applyQuery}
+        appliedQuery={appliedQuery}
+        filteredPeople={filteredPeople}
+        onSelected={selectPerson}
+        handleReset={handleReset}
+      />
```
Comment (dinaminahi) [inline src/App.tsx:60]: Do we need all these props& or we can create some handlers?

**Review by dinaminahi — CHANGES_REQUESTED** [review]
There is a bug we need to fix 

https://github.com/mate-academy/react_autocomplete/assets/46566190/9ceb190a-c8af-4635-bf9b-e0c4d74919bc

**Comment by Shevtsooov** [comment]
@dinaminahi Hello, thanks for the detailed review.
I have fixed the bug and added handlers in my `Drodown` component, still I am not sure how to reduce the number of props.
Please suggest a solution.

File: src/components/Dropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      </div>
+
+      {query && query === appliedQuery && (
+        <div className="dropdown-menu" role="menu">
+          <div className="dropdown-content">
+            {filteredPeople.length > 0 && (
+              filteredPeople.map(person => (
+                <div className="dropdown-item">
+                  {// eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-noninteractive-element-interactions
+                    <p
```
Comment (dffUqp) [inline src/components/Dropdown.tsx:65]: let's fix this eslint issue:smiley: 

I think we can replace `<p>` with `<a>`

File: src/components/Dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+  return (
+    <div className="dropdown is-active">
+      <div className="dropdown-trigger">
+        <input
+          type="text"
+          placeholder="Enter a part of the name"
+          className="input"
+          value={query}
+          onChange={(event) => handleSearch(event)}
```
Comment (dffUqp) [inline src/components/Dropdown.tsx:45]:
```suggestion
          onChange={handleSearch}
```

File: src/components/Dropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            alt="reset"
+            onClick={handleReset}
+          />
+        )}
+      </div>
+
+      {query && query === appliedQuery && (
+        <div className="dropdown-menu" role="menu">
+          <div className="dropdown-content">
+            {filteredPeople.length > 0 && (
```
Comment (dffUqp) [inline src/components/Dropdown.tsx:61]: do we really need this check?

**Review by dffUqp — APPROVED** [review]
GJ✔️

about props i think in you `App.tsx` you need only `selectedPerson` so the rest of the variables you can move to `DropDown.tsx`

**Pull request closed without merging**